test/core/dht.spec.js

```diff
@@ -7,45 +7,95 @@ const dirtyChai = require('dirty-chai')
 const expect = chai.expect
 chai.use(dirtyChai)
 
+const isNode = require('detect-node')
+
 const IPFSFactory = require('ipfsd-ctl')
 const IPFS = require('../../src/core')
 
 describe('dht', () => {
-  let ipfsd, ipfs
+  describe('enabled', () => {
+    let ipfsd, ipfs
+
+    before(function (done) {
+      this.timeout(30 * 1000)
 
-  before(function (done) {
-    this.timeout(30 * 1000)
+      const factory = IPFSFactory.create({ type: 'proc' })
 
-    const factory = IPFSFactory.create({ type: 'proc' })
+      factory.spawn({
+        exec: IPFS,
+        initOptions: { bits: 512 },
+        config: {
+          Bootstrap: []
+        }
+      }, (err, _ipfsd) => {
+        expect(err).to.not.exist()
+        ipfsd = _ipfsd
+        ipfs = _ipfsd.api
+        done()
+      })
+    })
 
-    factory.spawn({
-      exec: IPFS,
-      initOptions: { bits: 512 },
-      config: {
-        Bootstrap: []
+    after((done) => {
+      if (ipfsd) {
+        ipfsd.stop(done)
+      } else {
+        done()
       }
-    }, (err, _ipfsd) => {
-      expect(err).to.not.exist()
-      ipfsd = _ipfsd
-      ipfs = _ipfsd.api
-      done()
     })
-  })
 
-  after((done) => {
-    if (ipfsd) {
-      ipfsd.stop(done)
-    } else {
-      done()
-    }
+    describe('findprovs', () => {
+      it('should callback with error for invalid CID input', (done) => {
+        ipfs.dht.findProvs('INVALID CID', (err) => {
+          expect(err).to.exist()
+          expect(err.code).to.equal('ERR_INVALID_CID')
+          done()
+        })
+      })
+    })
   })
 
-  describe('findprovs', () => {
-    it('should callback with error for invalid CID input', (done) => {
-      ipfs.dht.findProvs('INVALID CID', (err) => {
-        expect(err).to.exist()
-        expect(err.code).to.equal('ERR_INVALID_CID')
+  describe('disabled in browser', () => {
+    if (isNode) { return }
+
+    let ipfsd, ipfs
+
+    before(function (done) {
+      this.timeout(30 * 1000)
+
+      const factory = IPFSFactory.create({ type: 'proc' })
+
+      factory.spawn({
+        exec: IPFS,
+        initOptions: { bits: 512 },
+        config: {
+          Bootstrap: []
+        }
+      }, (err, _ipfsd) => {
+        expect(err).to.not.exist()
+        ipfsd = _ipfsd
+        ipfs = _ipfsd.api
+        done()
+      })
+    })
+
+    after((done) => {
+      if (ipfsd) {
+        ipfsd.stop(done)
+      } else {
         done()
+      }
+    })
+
+    describe('put', () => {
+      it('should callback with error for DHT not available', async () => {
+        let res
+        try {
+          res = await ipfs.dht.put(Buffer.from('a'), Buffer.from('b'))
+        } catch (err) {
+          expect(err).to.exist()
```

Comment:
Is there an error code or message we can assert on?

Reply:
No! But, I can add it to `js-libp2p`. However, we will have to bump the `rc`. What do you think?

Reply:
We need to ensure the error is because the DHT is disabled and not because of anything else. Whatever you can do to make that happen is good 👍